test/token/ERC20WithPermit.test.js

```diff
@@ -0,0 +1,1069 @@
+              .permit(
+                permittingHolder.address,
+                anotherAccount.address,
+                allowance,
+                deadline,
+                signature.v,
+                signature.r,
+                signature.s
+              )
+
+            expect(
+              await token.allowance(
+                permittingHolder.address,
+                anotherAccount.address
+              )
+            ).to.equal(allowance)
+          })
+        })
+
+        context("when the spender had an approved amount", () => {
+          beforeEach(async () => {
+            const deadline = tomorrow
+            const initialAllowance = allowance.sub(10)
+            const signature = await getApproval(
+              initialAllowance,
+              anotherAccount.address,
+              deadline
+            )
+
+            await token
+              .connect(anotherAccount)
+              .permit(
+                permittingHolder.address,
+                anotherAccount.address,
+                initialAllowance,
+                deadline,
+                signature.v,
+                signature.r,
+                signature.s
+              )
+          })
+
+          it("should approve the requested amount and replaces the previous one", async () => {
+            const deadline = tomorrow
+            const signature = await getApproval(
+              allowance,
+              anotherAccount.address,
+              deadline
+            )
+
+            await token
+              .connect(anotherAccount)
+              .permit(
+                permittingHolder.address,
+                anotherAccount.address,
+                allowance,
+                deadline,
+                signature.v,
+                signature.r,
+                signature.s
+              )
+
+            expect(
+              await token.allowance(
+                permittingHolder.address,
+                anotherAccount.address
+              )
+            ).to.equal(allowance)
+          })
+        })
+      })
+    })
+
+    context("when the spender is the zero address", () => {
+      const allowance = permittingHolderBalance
+      it("should revert", async () => {
+        const deadline = tomorrow
+        const signature = await getApproval(allowance, ZERO_ADDRESS, deadline)
+
+        await expect(
+          token
+            .connect(anotherAccount)
+            .permit(
+              permittingHolder.address,
+              ZERO_ADDRESS,
+              allowance,
+              deadline,
+              signature.v,
+              signature.r,
+              signature.s
+            )
+        ).to.be.revertedWith("Approve to the zero address")
+      })
+    })
+
+    context("when given never expiring permit", () => {
+      // uint(-1)
+      const allowance = ethers.BigNumber.from(
+        "115792089237316195423570985008687907853269984665640564039457584007913129639935"
+      )
```

Comment:
Can we extract this value to a `MAX_UINT256` constant in `contract-test-helpers.js`?

Reply:
We can but I I think it will be used only in this single test.

Reply:
This can also serve as a helper that can be used in the other tests or repos, as this is the purpose of this repository. Plus having `MAX_UINT256` in the test instead of `115792089237316195423570985008687907853269984665640564039457584007913129639935` makes it clear what the value is.

Reply:
https://github.com/thesis/solidity-contracts/pull/3/commits/ee42c58081a345f759f8f89b7022c0f70109f14d